modules/cudacodec/test/test_video.cpp

```diff
@@ -344,6 +351,22 @@ CUDA_TEST_P(CheckDecodeSurfaces, Reader)
         for (int i = 0; i < 100; i++) ASSERT_TRUE(reader->grab());
     }
 }
+
+CUDA_TEST_P(CheckInitParams, Reader)
```

Comment:
The test is implemented here.  As mentioned in the description it currently only checks that the new parameters have been set.  To test the functionality an rstp streamer would be required.